@etobella 

We have identified this issue when migrating the module l10n_es_facturae_face. When trying to access an invoice whose customer has the value "FACe" set in the field l10n_es_facturae_sending_code, a frontend error occurs.

![edi_oca_error_1](https://github.com/user-attachments/assets/87285e9a-ca54-4a55-880d-604a0dabc568)

Complete error:

```
OwlError: An error occured in the owl lifecycle (see this Error's "cause" property)
    Error: An error occured in the owl lifecycle (see this Error's "cause" property)
        at handleError (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:916:101)
        at App.handleError (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1545:29)
        at Fiber._render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:941:19)
        at Fiber.render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:939:6)
        at ComponentNode.initiateRender (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1007:47)

Caused by: TypeError: Cannot read properties of undefined (reading 'name')
    at Field.template (eval at compile (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1503:374), <anonymous>:16:92)
    at Fiber._render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:940:96)
    at Fiber.render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:939:6)
    at ComponentNode.initiateRender (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1007:47)
```

This error is solved by the modifying the file /static/src/js/widget_edi.esm.js. The following lines

```
EdiConfigurationWidget.template = "edi_oca.EdiConfigurationWidget";

registry.category("fields").add("edi_configuration", EdiConfigurationWidget);
```

need to be replaced by

```
EdiConfigurationWidget.template = "edi_oca.EdiConfigurationWidget";

registry.category("fields").add("edi_configuration", {
    component: EdiConfigurationWidget,
});
```

It is not enough with those changes though, as another error occurs:

![edi_oca_error_2](https://github.com/user-attachments/assets/c7c59518-1cfd-4432-aa33-2e0135ec5478)

Detailed error:

```
OwlError: Invalid loop expression: "undefined" is not iterable
    Error: Invalid loop expression: "undefined" is not iterable
        at prepareList (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1120:12)
        at EdiConfigurationWidget.template (eval at compile (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1503:374), <anonymous>:14:54)
        at Fiber._render (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:940:96)
        at Fiber.render (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:939:6)
        at ComponentNode.initiateRender (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1007:47)
```

It is because the component template is not properly defined. That is why the following function has been added to /static/src/js/widget_edi.esm.js file.

```
    getInfo() {
        return this.props.record.data.edi_config
    }
```
The way data is retrieved from the template is also modified. These lines

```
        <t t-foreach="props.value" t-as="rule_id" t-key="rule_index">
            <t t-set="rule" t-value="props.value[rule_id]" />
```

has been replaced by the following:

```
        <t t-foreach="this.getInfo()" t-as="rule_id" t-key="rule_index">
            <t t-set="rule" t-value="this.getInfo()[rule_id]" />
```

--- a/edi_oca/README.rst
+++ b/edi_oca/README.rst
@@ -171,6 +171,7 @@ Contributors
 
 -  Simone Orsi <simahawk@gmail.com>
 -  Enric Tobella <etobella@creublanca.es>
+-  Manuel Regidor <manuel.regidor@sygel.es>
 
 Maintainers
 -----------

--- a/edi_oca/readme/CONTRIBUTORS.md
+++ b/edi_oca/readme/CONTRIBUTORS.md
@@ -1,2 +1,3 @@
 - Simone Orsi \<<simahawk@gmail.com>\>
 - Enric Tobella \<<etobella@creublanca.es>\>
+- Manuel Regidor \<<manuel.regidor@sygel.es>\>

--- a/edi_oca/static/description/index.html
+++ b/edi_oca/static/description/index.html
@@ -8,11 +8,10 @@
 
 /*
 :Author: David Goodger (goodger@python.org)
-:Id: $Id: html4css1.css 9511 2024-01-13 09:50:07Z milde $
+:Id: $Id: html4css1.css 8954 2022-01-20 10:10:25Z milde $
 :Copyright: This stylesheet has been placed in the public domain.
 
 Default cascading style sheet for the HTML output of Docutils.
-Despite the name, some widely supported CSS2 features are used.
 
 See https://docutils.sourceforge.io/docs/howto/html-stylesheets.html for how to
 customize this style sheet.
@@ -275,7 +274,7 @@ pre.literal-block, pre.doctest-block, pre.math, pre.code {
   margin-left: 2em ;
   margin-right: 2em }
 
-pre.code .ln { color: gray; } /* line numbers */
+pre.code .ln { color: grey; } /* line numbers */
 pre.code, code { background-color: #eeeeee }
 pre.code .comment, code .comment { color: #5C6576 }
 pre.code .keyword, code .keyword { color: #3B0D06; font-weight: bold }
@@ -301,7 +300,7 @@ span.option {
 span.pre {
   white-space: pre }
 
-span.problematic, pre.problematic {
+span.problematic {
   color: red }
 
 span.section-subtitle {
@@ -522,14 +521,13 @@ If you spotted it first, help us to smash it by providing a detailed and welcome
 <ul class="simple">
 <li>Simone Orsi &lt;<a class="reference external" href="mailto:simahawk&#64;gmail.com">simahawk&#64;gmail.com</a>&gt;</li>
 <li>Enric Tobella &lt;<a class="reference external" href="mailto:etobella&#64;creublanca.es">etobella&#64;creublanca.es</a>&gt;</li>
+<li>Manuel Regidor &lt;<a class="reference external" href="mailto:manuel.regidor&#64;sygel.es">manuel.regidor&#64;sygel.es</a>&gt;</li>
 </ul>
 </div>
 <div class="section" id="maintainers">
 <h2><a class="toc-backref" href="#toc-entry-14">Maintainers</a></h2>
 <p>This module is maintained by the OCA.</p>
-<a class="reference external image-reference" href="https://odoo-community.org">
-<img alt="Odoo Community Association" src="https://odoo-community.org/logo.png" />
-</a>
+<a class="reference external image-reference" href="https://odoo-community.org"><img alt="Odoo Community Association" src="https://odoo-community.org/logo.png" /></a>
 <p>OCA, or the Odoo Community Association, is a nonprofit organization whose
 mission is to support the collaborative development of Odoo features and
 promote its widespread use.</p>

--- a/edi_oca/static/src/js/widget_edi.esm.js
+++ b/edi_oca/static/src/js/widget_edi.esm.js
@@ -23,8 +23,13 @@ export class EdiConfigurationWidget extends Component {
         );
         this.action.doAction(action);
     }
+    get info() {
+        return this.props.record.data[this.props.name];
+    }
 }
 
 EdiConfigurationWidget.template = "edi_oca.EdiConfigurationWidget";
 
-registry.category("fields").add("edi_configuration", EdiConfigurationWidget);
+registry.category("fields").add("edi_configuration", {
+    component: EdiConfigurationWidget,
+});

--- a/edi_oca/static/src/xml/widget_edi.xml
+++ b/edi_oca/static/src/xml/widget_edi.xml
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <templates id="template" xml:space="preserve">
     <div t-name="edi_oca.EdiConfigurationWidget">
-        <t t-foreach="props.value" t-as="rule_id" t-key="rule_index">
-            <t t-set="rule" t-value="props.value[rule_id]" />
+        <t t-foreach="info" t-as="rule_id" t-key="rule_index">
+            <t t-set="rule" t-value="info[rule_id]" />
             <t t-if="rule.form">
                 <t t-set="btn" t-value="rule.form.btn" />
                 <button

